iiwa_driver/src/iiwa.cpp

```diff
@@ -218,27 +219,29 @@ namespace iiwa_ros {
             _controller_manager->update(ros::Time::now(), elapsed_time);
             _write(elapsed_time);
 
-            // publish additional outputs
-            if (_additional_pub.trylock()) {
-                _additional_pub.msg_.header.stamp = ros::Time::now();
-                for (unsigned i = 0; i < _num_joints; i++) {
-                    _additional_pub.msg_.external_torques.data[i] = _robot_state.getExternalTorque()[i];
-                    _additional_pub.msg_.commanded_torques.data[i] = _robot_state.getCommandedTorque()[i];
-                    _additional_pub.msg_.commanded_positions.data[i] = _robot_state.getCommandedJointPosition()[i];
-                }
-                _additional_pub.unlockAndPublish();
-            }
-
             _publish();
             rate.sleep();
         }
     }
 
     void Iiwa::_publish()
     {
-        std_msgs::Bool msg;
-        msg.data = _commanding;
-        _commanding_status_pub.publish(msg);
+        // publish commanding status
+        if (_commanding_status_pub.trylock()) {
+            _commanding_status_pub.msg_.data = _commanding;
+            _commanding_status_pub.unlockAndPublish();
+        }
+
+        // publish additional outputs
+        if (_additional_pub.trylock()) {
+            _additional_pub.msg_.header.stamp = ros::Time::now();
+            for (unsigned i = 0; i < _num_joints; i++) {
+                _additional_pub.msg_.external_torques.data[i] = _robot_state.getExternalTorque()[i];
+                _additional_pub.msg_.commanded_torques.data[i] = _robot_state.getCommandedTorque()[i];
+                _additional_pub.msg_.commanded_positions.data[i] = _robot_state.getCommandedJointPosition()[i];
+            }
+            _additional_pub.unlockAndPublish();
+		}
```

Comment:
Can you format the code better here? I use this [clang-format](https://github.com/resibots/format_code/blob/master/.clang-format) style.

Reply:
That was a "tabs and spaces" issue. Not sure where the tabs came in.

clang-format: I guess there's a way to specify one per project, but my environment defaults to the ROS C++ style and that one differs from what you are using.